### DISCUSSION TITLE
Restricting a GitHub Actions workflow to specific users

### DISCUSSION THREAD

**Discussion by alex-anson** [community/community#27947] [Actions] [answered]
Is it possible to restrict a GitHub Actions workflow to certain users?

Our current workaround is to use a protected branch, allowing workflows to trigger off of a push to that branch.

We want to be able to make a `package.json` command that deploys to production, only allowing it to be (successfully) used by selected individuals. Can we make a "push button" action using the `workflow_dispatch:` command to accomplish this?

Thank you for your time!

**Comment by d4rkn3m3s1s** [comment]


**Reply by alex-anson** [reply]
Awesome! How? I haven't been able to find anything in the docs...

**Comment by jsoref** [comment, marked as answer]
https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-51

sender | object | The user that triggered the event.
-- | -- | --

You can check the sender object against `id`s or `node_id`s...

**Reply by alex-anson** [reply]
Sweet!! Thank you! 🙏

**Reply by shaikhspear16** [reply]
Hey @alex-anson how did you end up doing it? Did you have to hardcode user names in your codebase and validate against the sender?

**Reply by singhravjot** [reply]
@alex-anson same query, how do implement it?

**Reply by alex-anson** [reply]
we didn't end up doing it. higher priority stuff got in the way... never revisited this

**Comment by jeremy-page** [comment]
This is not a very useful answer

**Comment by jo-elimu** [comment]
Is it possible to restrict workflows to `.github/CODEOWNERS` only?

**Reply by jsoref** [reply]
Workflows can run code that you write. https://github.com/marketplace/actions/codeowners-action

**Comment by martinmckenna** [comment]
This would be great if it was a setting. I want to use `workflow_dispatch` and filter who can/cannot run it manually https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_dispatch
